src/Nikolaev/Task1/MainWindow.cpp

```diff
@@ -0,0 +1,141 @@
+#include "MainWindow.h"
+
+#include <QMouseEvent>
+
+#include <cmath>
+
+#include <QScreen>
+
+void MainWindow::mousePressEvent(QMouseEvent *e)
+{
+    // Save mouse press position
+    mousePressPosition = QVector2D(e->localPos());
+}
+
+void MainWindow::mouseReleaseEvent(QMouseEvent *e)
+{
+    // Mouse release position - mouse press position
+    QVector2D diff = QVector2D(e->localPos()) - mousePressPosition;
+
+    // Rotation axis is perpendicular to the mouse position difference
+    // vector
+    QVector3D n = QVector3D(diff.y(), diff.x(), 0.0).normalized();
```

Comment:
```suggestion
    const auto n = QVector3D(diff.y(), diff.x(), 0.0).normalized();
```

Пробегитесь по коду, поправьте все такие места :)